code/modules/projectiles/projectile.dm

```diff
@@ -908,6 +910,9 @@
 				qdel(src)
 			return //Kill!
 		pixel_move(1, TRUE)
+		// No kevinz I do not care that this is a hitscan weapon, it is not allowed to travel 100 turfs in a tick
+		if(CHECK_TICK && QDELETED(src))
```

Comment:
I'm confused by this, should it be this instead?
```suggestion
		if(CHECK_TICK || QDELETED(src))
```

Reply:
nvm figured it out, I think you should put a comment explaining what this is doing though

Reply:
CHECK_TICK returns TRUE if it yields, FALSE otherwise. This can be read as "if we yield, and after yielding we are deleted, exit"
Can add a comment if you like

Reply:
could always do the nullcheck though, I don't think it's necessarily computationally expensive

Reply:
I'm not checking for null what do you mean?

Reply:
QDELETED macro has `isnull(x)` in it